### DISCUSSION TITLE
Error with ADComputeFiniteStrain

### DISCUSSION THREAD

**Discussion by rh201** [idaholab/moose#23495] [Q&A General] [answered]
Hi MOOSEer,

When I use the module [ADComputeFiniteStrain], the following error comes up. I checked the mesh and don't think the elements are heavily distorted, the strain level is low actually. Any suggestions/comments to avoid this error? Thanks.

`Cannot take square root of a number less than or equal to zero in the calculation of C1 for the Rashid approximation for the rotation tensor. This zero or negative number may occur when elements become heavily distorted.
Cannot take square root of a number less than or equal to zero in the calculation of C3_test for the Rashid approximation for the rotation tensor. This zero or negative number may occur when elements become heavily distorted.
`

**Comment by GiudGiud** [comment, marked as answer]
Hello

Did the solve finish after this?
You need to examine the output after this warning to see distortion. There is a FAILED execution flag for output you can use

The solution for this is usually to take smaller time steps, or if it's happening near a Dirichlet boundary condition, to not pre-set the dirichlet BCs. 

Guillaume

**Reply by rh201** [reply]
The solve stops calculation until reach the time limit, and the distortion is not big at all. And it's not happening near a Dirichlet B.C., I'll try with a smaller dt.

Here is the error output message.
[err.log](https://github.com/idaholab/moose/files/10794277/err.log)

**Reply by GiudGiud** [reply]
does it make the first time step?
The elements seem very inverted from the libmesh output in the log

**Reply by rh201** [reply]
No, it occurs in the middle fo the calculation. 
The mesh type is tetrahedron, does it matter?

**Reply by GiudGiud** [reply]
Tets should be fine.
